### ISSUE TITLE
Using SLURM for Imagenet training on multiple nodes

### ISSUE THREAD

**Issue by mesllo** [pytorch/examples#1012]
In the pytorch imagenet example of this repo, it says that for multiple nodes we have to run the command on each node like below:

![image](https://user-images.githubusercontent.com/10924797/173546864-66c56fa9-3aef-4f26-9e06-12866db2220f.png)

Since I am using a shared HPC cluster with SLURM, I cannot actively know which nodes my training will use so I'm not sure how to run these two commands. How can I run these two commands on the separate nodes using SLURM?

**Comment by BIGBALLON** [comment]
Hi, @mesllo, you can check https://github.com/BIGBALLON/distribuuuu for both ``launch`` and ``Slurm`` usages.

Please check [tutorial](https://github.com/BIGBALLON/distribuuuu/blob/master/tutorial) for detailed Distributed Training tutorials:

- Single Node Single GPU Card Training [[snsc.py](https://github.com/BIGBALLON/distribuuuu/blob/master/tutorial/snsc.py)]
- Single Node Multi-GPU Cards Training (with DataParallel) [[snmc_dp.py](https://github.com/BIGBALLON/distribuuuu/blob/master/tutorial/snmc_dp.py)]
- Multiple Nodes Multi-GPU Cards Training (with DistributedDataParallel)
torch.distributed.launch [[mnmc_ddp_launch.py](https://github.com/BIGBALLON/distribuuuu/blob/master/tutorial/mnmc_ddp_launch.py)]
torch.multiprocessing [[mnmc_ddp_mp.py](https://github.com/BIGBALLON/distribuuuu/blob/master/tutorial/mnmc_ddp_mp.py)]
- Slurm Workload Manager [[mnmc_ddp_slurm.py](https://github.com/BIGBALLON/distribuuuu/blob/master/tutorial/mnmc_ddp_slurm.py)]
- ImageNet training example [[imagenet.py](https://github.com/BIGBALLON/distribuuuu/blob/master/tutorial/imagenet.py)]

**Comment by msaroufim** [comment]
Hi @mesllo typically to avoid the need to ssh and copy over data to each node it's worth using a job scheduler. A good option is torchx which has builtin support for slurm https://github.com/pytorch/torchx - cc @d4l3k
